src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -182,14 +241,15 @@ func (r *StatefulSetResource) rollingUpdate(
 		volumes[vol.Name] = new(interface{})
 	}
 
-	for i := range podList.Items {
-		pod := podList.Items[i]
+	for i := range items {
+		pod := items[i]
 
-		if err = r.podEviction(ctx, &pod, &artificialPod, volumes); err != nil {
+		if err = r.podEviction(ctx, pod, &artificialPod, volumes); err != nil {
+			log.V(logger.DebugLevel).Error(err, "podEviction", "pod name", pod.Name)
```

Comment:
It's a debug log. The reconciliation log comes far enough up that it makes tracing behavior difficult.

Reply:
It's only 4 call stack apart. You can see that in
https://github.com/redpanda-data/redpanda/blob/5f9e5bf7736ddba99facda2b6c89953d9c6263f0/src/go/k8s/controllers/redpanda/cluster_controller.go#L1159

it will be printed by attachedResources. So in the end 3 times:
* this place
* attached resources
* controller runtime

I don't see benefit of having it here. I see only downside that it will cutter log output of cluster controller.

Reply:
It's a debug log. If you don't want to see it, don't enable debugging. When I was debugging this function, I needed this output.